File: store/model/comment.go
package model

import "time"

// Comment 评论模型
type Comment struct {
	Id        int    // PK
	PostId    int    // 文章PID
	Pid       int    // 回复评论ID，顶层主评论为0
	Name      string // 名称
	Email     string // 邮箱（不对外展示，仅用于生成头像）
	Website   string // 网址
	Content   string // 内容
	ReplyName string // 被回复人昵称，回复的回复时用于平铺展示"A 回复 B"
	IP        string // IP
	CreatedAt time.Time
}

// CommentWithPost 评论关联文章信息，用于侧边栏最新评论和后台列表
type CommentWithPost struct {
	Comment
	PostTitle string // 关联文章标题
	PostUrl   string // 关联文章缩略名
}

File: mobile/Models/Comment.swift
// 评论相关模型

import Foundation

/// 评论（api v1）
struct Comment: Identifiable, Decodable {
    let id: Int
    let pid: Int
    let name: String
    let avatar: String
    let website: String
    let content: String
    let reply_name: String
    let created_at: String      // proto: string（RFC3339 格式）
}

/// 评论列表请求
struct CommentListRequest: Encodable {
    let post_id: Int
}

/// 评论列表响应
struct CommentListResponse: Decodable {
    let list: [Comment]
}

/// 创建评论请求
struct CommentCreateRequest: Encodable {
    let post_id: Int
    let name: String
    let email: String?
    let website: String?
    let pid: Int
    let reply_name: String?
    let content: String
}

/// 创建评论响应
struct CommentCreateResponse: Decodable {
    let id: Int
}

/// 管理端评论
struct AdminComment: Identifiable, Decodable {
    let id: Int
    let post_id: Int
    let pid: Int
    let name: String
    let email: String
    let website: String
    let content: String
    let reply_name: String
    let ip: String
    let created_at: String      // proto: string（RFC3339 格式）
    let post_title: String
    let post_url: String
}

/// 管理端评论列表响应
struct AdminCommentListResponse: Decodable {
    let list: [AdminComment]
    let total: Int
}
